include/NovelRT/Ecs/SystemScheduler.h

```diff
@@ -32,6 +32,7 @@ namespace NovelRT::Ecs
 
         std::vector<Atom> _threadWorkItem;
         std::vector<std::thread> _threadCache;
+        std::vector<std::unique_ptr<tbb::mutex>> _mutexCache;
```

Comment:
IMO we should probably combine this with the threadAvailabilityMap TBH, something like this:

```cpp
class worker
{
    tbb::mutex sleep_blocker;
    std::atomic_bool _available;
};

std::vector<worker> _workers;
```

It may not be as efficient as the current solution, but it means we're not splitting the same information across multiple types, and then finding an available thread is effectively this:
```
for (auto& system : systems)
{
    auto available = [](const worker& worker){ return worker._available.load(/* some appropriate memory order */); }
    std::partition(_workers.begin(), _workers.end(), available);
    auto& worker = std::find_if(_workers.begin(), _workers.end(), worker);
    if (worker != _workers.end())
    {
        worker.workAvailable(system);
    }
}
```